test/Engine/ProtoTest/Associative/MethodResolution.cs

```diff
@@ -313,5 +313,57 @@ public void TestMethodResultionForNonExistType()
             thisTest.VerifyRunScriptSource(code, error);
             thisTest.Verify("z", null);
         }
+

```

Comment:
1. Should the test have a replication guide? Because Case 1a seems to be explicit about it
2. Does it necessarily have to test resolution on member functions?